### DISCUSSION TITLE
Redirection from Other server to Wiki js Server Using token

### DISCUSSION THREAD

**Discussion by gautamgupta12345** [requarks/wiki#6899] [Help / Questions]
We have our own website, we want to redirect from there to Wiki js Server Using the token, 
We got the token from Graphql API, but couldn't to redirect to wiki js server (Without login Since we already have token).

**Comment by rrg92** [comment]
Try setting your webserver to create a cookie called jwt with that token. But is your website and wiki arr different domains, this can be more complex, since you cannot set cookies to other domains.

**Reply by rrg92** [reply]
You can configure your wiki to be served through a proxy server like nginx. The access flow would be as follows:

```mermaid
graph LR
    A[Client] -->|http://yourwiki| B
    B -->|Contains| C[nginx]
    C -->|request| D[wiki]

    subgraph B[http://yourwiki]
    C
    D
    end

```



To facilitate this, create a "redirect" HTML page. This page should display the text "Redirecting..." and include a simple JavaScript script. The script activates upon loading and processes a query string (e.g., `http://yourwiki/redirect?page=xxx&auth=TOKEN`). The query string will contain the target page path in the wiki (`page`) and an authentication token (`auth`).

Consider adding encryption or encoding to the token to prevent it from being exposed and stored in the browser's history.

The JavaScript should set a cookie named "jwt" with the provided token, a method used by Wiki.js. After setting the cookie, the script will redirect the user to the desired page within the wiki, as specified by the path parameter in the query string.


The flow with this additions would be:

```mermaid
graph LR
    Client--> YourSite[http://yoursite]
    YourSite -->|?token=X&path=Y| nginxRedirect
    

    subgraph nginx["nginx: http://yourwiki"]
    nginxRedirect -->|PROXY| Wiki
    end

    subgraph nginxRedirect["/redirect"]
    C["set-cookie jwt token"]
    end

```
